// @shopify/polaris/build/cjs/tokens/tokens.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var breakpoints = require('./token-groups/breakpoints.json.js');
var color_dark = require('./token-groups/color.dark.json.js');
var depth = require('./token-groups/depth.json.js');
var legacyTokens = require('./token-groups/legacy-tokens.json.js');
var color_light = require('./token-groups/color.light.json.js');
var motion = require('./token-groups/motion.json.js');
var shape = require('./token-groups/shape.json.js');
var spacing = require('./token-groups/spacing.json.js');
var typography = require('./token-groups/typography.json.js');
var zIndex = require('./token-groups/z-index.json.js');
var utilities = require('./utilities.js');

/**
 * Values to convert to CSS custom properties.
 * @example {background: '#000'} // --p-background: #000;
 */

const osColorSchemes = {
  light: 'light',
  dark: 'dark'
};
/**
 * Polaris color schemes and their associated color tokens.
 */

const colorSchemes = {
  light: color_light["default"],
  dark: color_dark["default"]
};
const tokens = {
  breakpoints: utilities.tokensToRems(breakpoints["default"]),
  colorSchemes,
  depth: depth["default"],
  legacyTokens: utilities.tokensToRems(legacyTokens["default"]),
  motion: motion["default"],
  shape: utilities.tokensToRems(shape["default"]),
  spacing: utilities.tokensToRems(spacing["default"]),
  typography: utilities.tokensToRems(typography["default"]),
  zIndex: zIndex["default"]
};

exports.osColorSchemes = osColorSchemes;
exports.tokens = tokens;
